[ycdaniel326] iP

ycdaniel326 opened the pull request.

ycdaniel326 committed:
Renamed the filename to be YC and changed the bot name to YC. Added codes for greeting and exitting

ycdaniel326 committed:
Added Level 1-1 Echo

ycdaniel326 committed:
Added the list to store users' commands and ability to list out the commands.

ycdaniel326 committed:
Created the class Task for user input

ycdaniel326 committed:
Replaced the command list array to be class Task. Added the function of marking and unmarking tasks.

ycdaniel326 committed:
Added displayWelcomeMessage and displayByeMessage for printing the welcome and exit messages

limzerui commented on src/main/java/YC.java:
```diff
@@ -0,0 +1,69 @@
+import java.util.Scanner;
```
leave a line after your import statements before the start of class declaration

limzerui commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,30 @@
+public class Task {
+    protected String description;
+    protected boolean isDone;
```
Could these be private to keep invariants local?

limzerui commented on src/main/java/YC.java:
```diff
@@ -0,0 +1,69 @@
+import java.util.Scanner;
+public class YC {
+    public static void main(String[] args) {
+        String bot_Name = "YC";
```
prefer camelCase without underscores (botName)

limzerui replied:
also consider private static final if constant
```suggestion
       private static final String BOT_NAME + "YC";
       ```

limzerui reviewed (commented):
code is readable.
suggested a few areas of improvement.

ycdaniel326 committed:
Added the classes Deadline, Event, Todo for handling the respective tasks entry.
Amended slightly the coding format.

ycdaniel326 committed:
Refactor and add separate methods for each cases at the end to make the if clauses more readable.

yentheng0110 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,34 @@
+    }
+
+    public String getStatusIcon() {
+        return (isDone ? "X" : " "); // mark done task with X
```
All comments in the code can be removed

yentheng0110 commented on src/main/java/YC.java:
```diff
@@ -0,0 +1,139 @@
+        String botName = "YC";
+        Scanner scanner = new Scanner(System.in);
+        String userCommand;
+        Task[] commandList = new Task[100];
```
100 here is a magic number, perhaps you can consider abstracting it out as a constant

yentheng0110 commented on src/main/java/YC.java:
```diff
@@ -0,0 +1,139 @@
+        System.out.println("\t" + "Hello! I'm " + botName);
+        System.out.println("\t" + "What can I do for you?");
+        System.out.println("\t" + "____________________________________________________________" + "\n");
+    }
```
I like how you abstract each function as a method, good job!

yentheng0110 commented on src/main/java/YC.java:
```diff
@@ -0,0 +1,139 @@
+    }
+
+    private static int ProcessTodo(String userCommand, Task[] commandList, int totalCommands) {
+        String description = userCommand.substring(5);
```
The number 5 is a magic number, consider abstracting it out as a constant. The same applies to other methods

yentheng0110 reviewed (commented):
In general, your code is clean with proper method abstraction, minor changes were suggested.

ycdaniel326 committed:
Created two classes of Exception to handle potential errors when running the code.

ycdaniel326 committed:
Added two more exception classes to handle invalid command or invalid index cases. Modified the message shown in EmptyEntryException

ycdaniel326 committed:
Added the error checking codes into the methods.
Also, modifed the codes such that it is more fitting the coding standard and more organised.

ycdaniel326 committed:
Merge branch 'branch-Level-5'

ycdaniel326 committed:
Refined the Task class to remove unnecessary codes

ycdaniel326 committed:
Added the packages Exceptions and Tasks to store the respective classes.

ycdaniel326 committed:
Merge branch 'branch-A-Packages'

ycdaniel326 committed:
Refactored the main code.

ycdaniel326 committed:
Changed the commandList to an array list.
Refined and made the commandList static so it can be accessed by the methods directly without needing to pass.
Amended the method codes to use the methods of array list.

ycdaniel326 committed:
Added the deleteTask method.
Renamed several methods to follow camelCase.

ycdaniel326 committed:
Added the methods saveTasks and loadTasks for level 7. Added the method toStorage for printing the displays of different task types.
(Remark: Since I have made the commandList a static array list in level 6, I have changed the format here as well to prevent merge conflicts)

ycdaniel326 committed:
Merge branch 'branch-Level-6'

ycdaniel326 committed:
Solved the merge conflicts.
Added level 7 methods, saveTasks and loadTasks, with conflicts solved.

ycdaniel326 committed:
Removed a repeated line that I overlooked previously

ycdaniel326 committed:
Created the jar for the project. This is the MANIFEST.MF

ycdaniel326 committed:
updated the event and deadline classes to ensure they are saved in correct format

ycdaniel326 committed:
Deleted the "Duke" file.

ycdaniel326 committed:
Created the classes Ui, Storage, Parser and TaskList and extracted out the codes.

ycdaniel326 committed:
Amended the YC.java accordingly after the extraction of classes to make it more OOP.

ycdaniel326 committed:
Fixed the "toStorage" method in Deadline and Event classes to ensure the saving and loading functions work in the correct format

ycdaniel326 committed:
Added the findTask method in Parser to handle the "find" command.

ycdaniel326 committed:
Added JavaDoc comments to the code for most public classes/methods as well as some non-trivial private methods

ycdaniel326 committed:
Added User Guide in README.md

ycdaniel326 committed:
Merge branch 'master' of https://github.com/ycdaniel326/ip

* 'master' of https://github.com/ycdaniel326/ip:
  Added User Guide in README.md

ycdaniel326 committed:
Added the UserGuide in README.md in docs

ycdaniel326 committed:
Merge branch 'master' of https://github.com/ycdaniel326/ip
